scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -1208,6 +1211,79 @@ static bool keep_running(struct cpu_ctx *cctx, struct task_struct *p)
 	return false;
 }
 
+bool timer_antistall(s32 cpu, struct cpu_ctx *cctx){
+	struct task_struct *p;
+	struct task_ctx *tctx;
+	u64 dsq_id, now;
+	u32 idx;
+	
+	if (!cctx || !cctx->layer_idx)
+		return false;
+
+	now = bpf_ktime_get_ns();
+	
+	if(!dsq_timer_check_flag)
+		return false;
+
+	bpf_for(idx, 0, nr_layers) {
+		dsq_id = rotate_layer_id(cctx->layer_idx, idx);
+
+		// check non-fallback dsqs.
+		bpf_for_each(scx_dsq, p, dsq_id, 0) {
+
+			if ((tctx = lookup_task_ctx(p))) {
+				if(tctx->running_at) {
+					if(now - tctx->running_at > NSEC_PER_MSEC * 5) {
```

Comment:
`running_at` is the last the the task got on the CPU. `p->scx.runnable_at` is the time when the task became runnable (ie. the task has been waiting to go on a CPU since this time) which is the right timestamp to calculate delta from. Note that `p->scx.runnable_at` is in jiffies. You can get the current jiffies with `bpf_jiffies64()` and the frequency of the current kernel with `extern unsigned CONFIG_HZ __kconfig;`.

Reply:
thanks!

Reply:
hmm this works locally, I think I need to edit a kconfig to make it work on ci (will figure it out and add it to PR):

```
09:50:59 [DEBUG] libbpf: object 'bpf_bpf': failed (-22) to create BPF token from '/sys/fs/bpf', skipping optional step...

09:50:59 [WARN] libbpf: failed to open system Kconfig
```